### PR TITLE
[Serializer] Remove unused GetSetMethodNormalizer::denormalize

### PR DESCRIPTION
**Pull request by dunglas**
| Q             | A
| ------------- | ---
| Branch?       | 3.1
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | n/a
| License       | MIT
| Doc PR        | n/a

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
:+1:

**Comment by stof** [comment]
Are you sure about this ?

**Comment by nicolas-grekas** [comment]
Here is the patch for this method comparing it to its parent implem.
Do you mean that this difference is useless and the parent implem is good enough?
```diff
     public function denormalize($data, $class, $format = null, array $context = array())
     {
-        if (!isset($context['cache_key'])) {
-            $context['cache_key'] = $this->getCacheKey($format, $context);
-        }
         $allowedAttributes = $this->getAllowedAttributes($class, $context, true);
         $normalizedData = $this->prepareForDenormalization($data);
 
         $reflectionClass = new \ReflectionClass($class);
         $object = $this->instantiateObject($normalizedData, $class, $context, $reflectionClass, $allowedAttributes);
 
+        $classMethods = get_class_methods($object);
         foreach ($normalizedData as $attribute => $value) {
             if ($this->nameConverter) {
                 $attribute = $this->nameConverter->denormalize($attribute);
             }
 
-            if (($allowedAttributes !== false && !in_array($attribute, $allowedAttributes)) || !$this->isAllowedAttribute($class, $attribute, $format, $context)) {
-                continue;
-            }
+            $allowed = $allowedAttributes === false || in_array($attribute, $allowedAttributes);
+            $ignored = in_array($attribute, $this->ignoredAttributes);
+
+            if ($allowed && !$ignored) {
+                $setter = 'set'.ucfirst($attribute);
 
-            $value = $this->validateAndDenormalize($class, $attribute, $value, $format, $context);
-            try {
-                $this->setAttributeValue($object, $attribute, $value, $format, $context);
-            } catch (InvalidArgumentException $e) {
-                throw new UnexpectedValueException($e->getMessage(), $e->getCode(), $e);
+                if (in_array($setter, $classMethods) && !$reflectionClass->getMethod($setter)->isStatic()) {
+                    $object->$setter($value);
+                }
             }
         }
```

**Comment by dunglas** [comment]
@nicolas-grekas @stof In fact the parent method is better because it contains cache and validation. This methods deactivate cache and validation and IMO should be removed.

File: src/Symfony/Component/Serializer/Normalizer/GetSetMethodNormalizer.php [hunk 1]
```diff
@@ -38,40 +38,6 @@ class GetSetMethodNormalizer extends AbstractObjectNormalizer
...
-        $object = $this->instantiateObject($normalizedData, $class, $context, $reflectionClass, $allowedAttributes);
-
-        $classMethods = get_class_methods($object);
-        foreach ($normalizedData as $attribute => $value) {
-            if ($this->nameConverter) {
-                $attribute = $this->nameConverter->denormalize($attribute);
-            }
-
-            $allowed = $allowedAttributes === false || in_array($attribute, $allowedAttributes);
-            $ignored = in_array($attribute, $this->ignoredAttributes);
```
Comment (xabbuh) [inline src/Symfony/Component/Serializer/Normalizer/GetSetMethodNormalizer.php]: If I didn't miss anything, ignored attributes will now no longer be ignored anymore.

**Review by xabbuh — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/Serializer/Normalizer/GetSetMethodNormalizer.php (hunk 1 above)
Comment (dunglas) [inline src/Symfony/Component/Serializer/Normalizer/GetSetMethodNormalizer.php]: It's handled here: https://github.com/dunglas/symfony/blob/0896b02be073b8fa60fe6e8b3cf5d4f81b748bf8/src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php#L217

Comment (xabbuh) [inline src/Symfony/Component/Serializer/Normalizer/GetSetMethodNormalizer.php]: thx

**Review by xabbuh — APPROVED** [review]

**Comment by theofidry** [comment]
@dunglas whilst I agree with the removal of the PR wouldn't that break Sf BC break policy as a user may potentially extend this serializer and use this method? Wouldn't it be better to just deprecate it for now?

**Comment by xabbuh** [comment]
@theofidry The method is still there (it's defined by the `AbstractObjectNormalizer`).

**Comment by theofidry** [comment]
Fair, 👍 then

**Comment by fabpot** [comment]
Thank you @dunglas.

**Pull request merged**